### PR TITLE
Added email validation tests for core.js

### PR DESCRIPTION
**Pull request by Supun94**
Pull Request for Issue # .

### Summary of Changes
This PR contains improvements of core.js JavaScript library. Adding tests for email validation.  

### Testing Instructions
None, only run on Travis and can test it locally.

### Documentation Changes Required
None

### CODE REVIEW

File: tests/javascript/core/spec.js [hunk 1]
```diff
@@ -241,4 +241,61 @@ define(['jquery', 'testsRoot/core/spec-setup', 'jasmineJquery'], function ($) {
...
+            {emailaddress: 'much.”more\ unusual”@example.com', validate: false},
+            {emailaddress: 'very.unusual.”@”.unusual.com@example.com', validate: false},
+            {
+                emailaddress: 'very.”(),:;<>[]”.VERY.”very@\\ "very”.unusual@strange.example.com',
+                validate: false,
+                message: 'is invalid'
+            },
+            {emailaddress: 'plainaddress', validate: false},
+            {emailaddress: '#@%^%#$@#$@#.com', validate: false},
+            {emailaddress: 'username@kasdkasd', validate: false},
```
Comment (brianteeman) [inline tests/javascript/core/spec.js]: this is a valid email address on a local system

Comment (yvesh) [inline tests/javascript/core/spec.js]: @brianteeman Not after the IETF standards and not on our validator we test here ;-)

File: tests/javascript/core/spec.js [hunk 2]
```diff
@@ -241,4 +241,61 @@ define(['jquery', 'testsRoot/core/spec-setup', 'jasmineJquery'], function ($) {
...
+            {emailaddress: '@example.com', validate: false},
+            {emailaddress: 'Your Name test@example.com', validate: false},
+            {emailaddress: 'email.example.com', validate: false},
+            {emailaddress: 'test@example@example.com', validate: false},
+            {emailaddress: '    .test@example.com', validate: false},
+            {emailaddress: 'email.@example.com', validate: true},
+            {emailaddress: 'email..test@example.com', validate: true},
+            {emailaddress: 'あいうえお@example.com', validate: false},
+            {emailaddress: 'test@example.com (Your Name)', validate: false},
+            {emailaddress: 'test@example', validate: false},
```
Comment (brianteeman) [inline tests/javascript/core/spec.js]: this is the same example as line 271 and is a valid email address on a local system

File: tests/javascript/core/spec.js (hunk 1 above)
Comment (brianteeman) [inline tests/javascript/core/spec.js]: not much point in having a validation test for an email address format that we allow ;)

Comment (brianteeman) [inline tests/javascript/core/spec.js]: because it is a perfectly valid email address

**Comment by brianteeman** [comment]
Is this the code that is being tested
https://github.com/joomla/joomla-cms/blob/staging/media/system/js/core-uncompressed.js#L230

Its not actually used anywhere

**Comment by dgt41** [comment]
@Supun94 according to the docblock this is only used in one place in Joomla:
```js
	/**
	 * USED IN: administrator/components/com_banners/views/client/tmpl/default.php
	 * Actually, probably not used anywhere. Can we deprecate in favor of <input type="email">?
	 *
	 * Verifies if the string is in a valid email format
	 *
	 * @param string
	 * @return boolean
	 */
	Joomla.isEmail = function( text ) {
		var regex = /^[\w.!#$%&‚Äô*+\/=?^`{|}~-]+@[a-z0-9-]+(?:\.[a-z0-9-]{2,})+$/i;
		return regex.test( text );
	};
```

Basically I would like to see all these very specific functions moving out of core.js in J4

But this is fine for Joomla 3.x

**Comment by brianteeman** [comment]
Its not even used there as that file referred to in the doc block doesnt exist

File: tests/javascript/core/spec.js (hunk 1 above)
Comment (brianteeman) [inline tests/javascript/core/spec.js]: I see now why we dont use it in the registration

File: tests/javascript/core/spec.js (hunk 2 above)
Comment (Supun94) [inline tests/javascript/core/spec.js]: @brianteeman Removed it :)

**Comment by roland-d** [comment]
Closing this as this code seems completely unused. Email checking is done in the file media/system/js/validate-uncompressed.js on line 213.

Thank you for your work on this.

**Pull request closed without merging**